robot/control/Src/modules/LEDModule.cpp

```diff
@@ -168,4 +198,52 @@ void LEDModule::setColor(uint32_t led0, uint32_t led1) {
     dotStarNCS.write(false);
     dotStarSPI.lock()->transmit(data);
     dotStarNCS.write(true);
-}
\ No newline at end of file
+}
+
+void LEDModule::displayErrors() {
+    // Check if there are errors to display, else exit
+    if (colorQueue.size() == 0) {
+        return;
+    }
+
+    if (lightsOn && framesOnCounter >= framesOn){
+        // If error lights have been on long enough, switch them off
+        framesOnCounter = 0;
+        lightsOn = false;
+        setColor(0x000000,0x000000, 0x000000);
+    } else if (!lightsOn && framesOffCounter >= framesOff) {
+        // If error lights have been off long enough, switch them back on to next error color
+        framesOffCounter = 0;
+        lightsOn = true;
+
+        // Make sure index loops back if it has reached end of vector
+        if (++index > colorQueue.size()-1) {
+            index = 0;
+        }
+        Error e = colorQueue.at(index);
+        setColor(e.led0, e.led1, e.led2);
+    } else {
+        // Increment frame counters for lights on/off
+        lightsOn ? framesOnCounter++ : framesOffCounter++;
+    }
+}
+
+void LEDModule::addError(Error newError) {
+    for(Error currentError: colorQueue) {
+        if (currentError.led0 == newError.led0 &&
+            currentError.led1 == newError.led1 &&
+            currentError.led2 == newError.led2) {
+            return;
+        }
+    }
+    colorQueue.push_back(newError);
+}
+
+void LEDModule::removeError(Error target) {
+    colorQueue.erase(std::remove_if(colorQueue.begin(), colorQueue.end(),
+            [&](Error err) {
+                return err.led0 == target.led0 &&
+                       err.led1 == target.led1 &&
+                       err.led2 == target.led2;
+           }), colorQueue.end());
```

Comment:
This makes me nervous in an embedded environment since I haven't tried any of these algorithms/lambdas in this context. Since this is getting checked every frame, it may cause some problems. It's definitely something to check

Reply:
I don't really see too much harm in this lambda because it's just simply checking whether given sets of constants are equals from different error structs. However, I'll make note to test this for safety